### PR TITLE
Accumulate the stored fees for blocks in the block state

### PR DESCRIPTION
**Pull request by gorillainduction**
This avoids recalculating the fees when giving the miner reward.

I'll rebase this PR and make it towards master after the merge.

https://www.pivotaltracker.com/n/projects/2124891/stories/158718222

### CODE REVIEW

**Comment by lucafavatella** [comment]
Dialyzer on CI [reported warning](https://circleci.com/gh/aeternity/epoch/19289?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) so I took the liberty of pushing a commit fixing that.

**Comment by gorillainduction** [comment]
Rebased on master and changed the base branch to master.

File: apps/aecore/src/aec_chain_state.erl [hunk 1]
```diff
@@ -447,75 +448,97 @@ assert_micro_signature(PrevNode, Node) ->
...
-             hash(Node)};
-        false -> db_find_state(prev_hash(Node))
+             #fork_info{ difficulty = aec_block_genesis:genesis_difficulty()
+                       , fork_id = hash(Node)
+                       , fees = 0
+                       }
+            };
+        false ->
+            PrevHash = prev_hash(Node),
+            case db_find_state(PrevHash) of
```
Comment (hanssv) [inline apps/aecore/src/aec_chain_state.erl]: Both db_find_state and db_get_node for same hash, synergy possible?

Comment (gorillainduction) [reply apps/aecore/src/aec_chain_state.erl]: It is from two different tables. Not much synergy to find there.

**Pull request merged**